transport/grpc/transport.go

```diff
@@ -85,6 +85,7 @@ func (t *Transport) NewInbound(listener net.Listener, options ...InboundOption)
 }
 
 // NewSingleOutbound returns a new Outbound for the given adrress.
+// Note: This does not support TLS. See TLS example in doc.go.
```

Comment:
nit: I'm sure it can be addressed in this diff, but it seems odd that NewOutbound supports TLS but NewSingleOutbound does not.  As a somewhat naive user I would expect the only difference between these two APIs is how peers are chosen.

Reply:
We can't alter the signature, but we should add a new analog method which does support DialOptions.

Reply:
We can add a `NewSingleOutbound` method to the dialer, but it would be awkward since we can’t use varargs for both DialOptions and OutboundOptions on the same function signature.

The naïve expectation is not so naïve—we’ve been surprised too. The NewSingleOutbound method on all of the transport protocols is a thin wrapper around peer.NewSingleChooser and hostport.Identify, but the convenience is darn nice. It might have been wiser to omit the convenience methods.